webapp/src/client.ts

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import manifest from './manifest';
+
+export interface ModerationStatusResponse {
+    enabled: boolean;
+    excluded: boolean;
+}
+
+export class Client {
+    private baseUrl: string;
+
+    constructor() {
+        this.baseUrl = `/plugins/${manifest.id}`;
+    }
+
+    async enableChannelModeration(channelId: string): Promise<void> {
```

Comment:
Since we aren't importing client4 here to handle our requests, how does the authorization token get set? We are using client4 in master but not in this branch

Reply:
@BenCookie95 good catch, we should absolutely be using Client4 here. It's getting the authorization token by manually setting `credentials: 'same-origin'` in the fetch options. I've switched this to use Client4, instead.